staging/kos/pkg/registry/network/subnet/strategy.go

```diff
@@ -66,48 +71,218 @@ func SelectableFields(obj *network.Subnet) fields.Set {
+		validationErrors = append(validationErrors, vniRangeErrors...)
+		validationErrors = append(validationErrors, cidrFormatErrors...)
+
+		// If we are here, both the VNI range and the CIDR format of the created subnet are invalid.
+		// All of the successive validation steps rely on at least one of these two values being valid,
+		// thus we return immediately
+		return validationErrors
+	}
+
+	// We need to check the under-validation subnet against subnets with the same VNI only. Other subnets can be ignored
+	allSubnetsWithSameVNI, err := s.allSubnetsWithVNI(thisSubnet.Spec.VNI)
 	if err != nil {
-		errs = append(errs, field.Invalid(field.NewPath("spec", "ipv4"), subnet.Spec.IPv4, fmt.Sprintf("CIDR parse error: %s", err.Error())))
+		// If we're here it was not possible to fetch the subnets with the same VNI as the one under creation,
+		// thus validation cannot proceed and we return immediately
+		// ? Consider adding a more informative error message?
+		validationErrors = append(validationErrors, field.InternalError(nil, fmt.Errorf("server internal error, try again later")))
+		return validationErrors
+	}
+
+	var sameVNISameNamespaceErrors field.ErrorList
+	if len(vniRangeErrors) == 0 {
+		sameVNISameNamespaceErrors = s.checkSameVNISameNamespace(thisSubnet, allSubnetsWithSameVNI)
+	}
+
+	var cidrDoesNotOverlapErrors field.ErrorList
+	if len(cidrFormatErrors) == 0 {
+		cidrDoesNotOverlapErrors = s.checkCIDRDoesNotOverlap(thisSubnet, allSubnetsWithSameVNI)
+	}
+
+	validationErrors = append(validationErrors, vniRangeErrors...)
+	validationErrors = append(validationErrors, cidrFormatErrors...)
+	validationErrors = append(validationErrors, sameVNISameNamespaceErrors...)
+	validationErrors = append(validationErrors, cidrDoesNotOverlapErrors...)
+
+	if len(validationErrors) > 0 {
+		return validationErrors
 	}
-	return errs
+
+	return nil
 }
 
-func (subnetStrategy) AllowCreateOnUpdate() bool {
+func (*subnetStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
-func (subnetStrategy) AllowUnconditionalUpdate() bool {
+func (*subnetStrategy) AllowUnconditionalUpdate() bool {
 	return false
 }
 
-func (subnetStrategy) Canonicalize(obj runtime.Object) {
+func (*subnetStrategy) Canonicalize(obj runtime.Object) {
+}
+
+func (s *subnetStrategy) ValidateUpdate(ctx genericapirequest.Context, obj, old runtime.Object) field.ErrorList {
+	if _, isSubnet := old.(*network.Subnet); !isSubnet {
+		// ? Consider adding a more informative error message?
+		// ? Consider invoking panic() instead. This is a severe bug that should never happen.
+		return field.ErrorList{field.InternalError(nil, fmt.Errorf("server internal error, try again later"))}
+	}
+	return s.Validate(ctx, obj)
+}
+
+// ! These two consts harm code reusability, because they're set to the OVS-specific values.
+// ! OVS is the low-level networking implementation used for this project, but this code
+// ! should not depend on OVS being used.
```

Comment:
Supporting generalization beyond OVS is, at present, a non-goal and involves changes throughout --- starting with the name (Kubernetes Ovs Sdn).